Broken installation of vkcube

cmake --install fails
```
juanramos@Juans-MBP KHR-Vulkan-Tools % cmake --install build --prefix /tmp
-- Install configuration: "Debug"
-- Up-to-date: /tmp/cube/vkcube.app
-- Up-to-date: /tmp/cube/vkcube.app/Contents
-- Up-to-date: /tmp/cube/vkcube.app/Contents/MacOS
-- Installing: /tmp/cube/vkcube.app/Contents/MacOS/vkcube
-- Up-to-date: /tmp/cube/vkcube.app/Contents/Resources
-- Up-to-date: /tmp/cube/vkcube.app/Contents/Resources/VulkanIcon.icns
-- Up-to-date: /tmp/cube/vkcube.app/Contents/Resources/Main.storyboardc
-- Up-to-date: /tmp/cube/vkcube.app/Contents/Resources/Main.storyboardc/MainMenu.nib
-- Up-to-date: /tmp/cube/vkcube.app/Contents/Resources/Main.storyboardc/XfG-lQ-9wD-view-m2S-Jp-Qdl.nib
-- Up-to-date: /tmp/cube/vkcube.app/Contents/Resources/Main.storyboardc/NSWindowController-B8D-0N-5wS.nib
-- Up-to-date: /tmp/cube/vkcube.app/Contents/Resources/Main.storyboardc/Info.plist
-- Up-to-date: /tmp/cube/vkcube.app/Contents/Resources/vulkan
-- Up-to-date: /tmp/cube/vkcube.app/Contents/Resources/vulkan/icd.d
-- Installing: /tmp/cube/vkcube.app/Contents/Resources/vulkan/icd.d/MoltenVK_icd.json
-- Up-to-date: /tmp/cube/vkcube.app/Contents/Frameworks
-- Up-to-date: /tmp/cube/vkcube.app/Contents/Frameworks/libMoltenVK.dylib
-- Up-to-date: /tmp/cube/vkcube.app/Contents/Info.plist
-- fixup_bundle
--   app='/tmp/cube/vkcube.app'
--   libs=''
--   dirs=''
--   ignoreItems=''
-- fixup_bundle: preparing...
-- 
warning: cannot resolve item '@rpath/libvulkan.1.dylib'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/libvulkan.1.dylib' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/libvulkan.1.dylib'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?
juanramos@Juans-MBP KHR-Vulkan-Tools % cmake --install build --prefix /tmp     
-- Install configuration: "Debug"
-- Up-to-date: /tmp/cube/vkcube.app
-- Up-to-date: /tmp/cube/vkcube.app/Contents
-- Up-to-date: /tmp/cube/vkcube.app/Contents/MacOS
-- Installing: /tmp/cube/vkcube.app/Contents/MacOS/vkcube
-- Up-to-date: /tmp/cube/vkcube.app/Contents/Resources
-- Up-to-date: /tmp/cube/vkcube.app/Contents/Resources/VulkanIcon.icns
-- Up-to-date: /tmp/cube/vkcube.app/Contents/Resources/Main.storyboardc
-- Up-to-date: /tmp/cube/vkcube.app/Contents/Resources/Main.storyboardc/MainMenu.nib
-- Up-to-date: /tmp/cube/vkcube.app/Contents/Resources/Main.storyboardc/XfG-lQ-9wD-view-m2S-Jp-Qdl.nib
-- Up-to-date: /tmp/cube/vkcube.app/Contents/Resources/Main.storyboardc/NSWindowController-B8D-0N-5wS.nib
-- Up-to-date: /tmp/cube/vkcube.app/Contents/Resources/Main.storyboardc/Info.plist
-- Up-to-date: /tmp/cube/vkcube.app/Contents/Resources/vulkan
-- Up-to-date: /tmp/cube/vkcube.app/Contents/Resources/vulkan/icd.d
-- Up-to-date: /tmp/cube/vkcube.app/Contents/Resources/vulkan/icd.d/MoltenVK_icd.json
-- Up-to-date: /tmp/cube/vkcube.app/Contents/Frameworks
-- Up-to-date: /tmp/cube/vkcube.app/Contents/Frameworks/libMoltenVK.dylib
-- Up-to-date: /tmp/cube/vkcube.app/Contents/Info.plist
-- fixup_bundle
--   app='/tmp/cube/vkcube.app'
--   libs=''
--   dirs=''
--   ignoreItems=''
-- fixup_bundle: preparing...
-- 
warning: cannot resolve item '@rpath/libvulkan.1.dylib'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/libvulkan.1.dylib' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/libvulkan.1.dylib'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- 
warning: cannot resolve item '@rpath/libvulkan.1.dylib'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

CMake Error at /opt/homebrew/Cellar/cmake/3.27.7/share/cmake/Modules/BundleUtilities.cmake:458 (message):
  otool -l failed: 1

  error:
  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/otool-classic:
  can't open file: @rpath/libvulkan.1.dylib (No such file or directory)

Call Stack (most recent call first):
  /opt/homebrew/Cellar/cmake/3.27.7/share/cmake/Modules/BundleUtilities.cmake:527 (get_item_rpaths)
  /opt/homebrew/Cellar/cmake/3.27.7/share/cmake/Modules/BundleUtilities.cmake:649 (set_bundle_key_values)
  /opt/homebrew/Cellar/cmake/3.27.7/share/cmake/Modules/BundleUtilities.cmake:933 (get_bundle_keys)
  build/cube/cmake_install.cmake:58 (fixup_bundle)
  build/cmake_install.cmake:47 (include)
```

Looks like this has been going on for a bit.

Note we have some questionable install code

```
    # Fix up the library references to be self-contained within the bundle.
    install(CODE "
        include(BundleUtilities)
        fixup_bundle(\${CMAKE_INSTALL_PREFIX}/cube/vkcube.app \"\" \"${Vulkan_LIBRARY_DIR}\")
        ")
```

Vulkan_LIBRARY_DIR doesn't exist.

--- a/.github/workflows/tools.yml
+++ b/.github/workflows/tools.yml
@@ -158,17 +158,13 @@ jobs:
             - name: Add ccache to PATH
               run: echo "/usr/lib/ccache:/usr/local/opt/ccache/libexec" >> $GITHUB_PATH
 
-            - name: Configure
-              run: cmake -S. -B build -G "Ninja" -DCMAKE_BUILD_TYPE=Release -D UPDATE_DEPS=ON -D INSTALL_ICD=ON -D BUILD_TESTS=ON -D ENABLE_ADDRESS_SANITIZER=ON -D BUILD_WERROR=ON
-              env:
-                MACOSX_DEPLOYMENT_TARGET: 10.15
+            - run: cmake -S. -B build -G Ninja -DCMAKE_BUILD_TYPE=Release -D UPDATE_DEPS=ON -D INSTALL_ICD=ON -D BUILD_TESTS=ON -D ENABLE_ADDRESS_SANITIZER=ON -D BUILD_WERROR=ON
 
-            - name: Build
-              run: cmake --build build
+            - run: cmake --build build
 
-            - name: Test
-              working-directory: ./build
-              run: ctest --output-on-failure
+            - run: ctest --output-on-failure --test-dir build/
+
+            - run: cmake --install build --prefix /tmp
 
     android:
       runs-on: ubuntu-22.04

--- a/cube/CMakeLists.txt
+++ b/cube/CMakeLists.txt
@@ -255,14 +255,15 @@ target_compile_definitions(vkcube PRIVATE VK_NO_PROTOTYPES)
 if (ANDROID)
     install(TARGETS vkcube DESTINATION ${CMAKE_INSTALL_LIBDIR})
 elseif(APPLE)
-    # Keep RPATH so fixup_bundle can use it to find libraries
-    set_target_properties(vkcube PROPERTIES INSTALL_RPATH_USE_LINK_PATH TRUE)
-    install(TARGETS vkcube BUNDLE DESTINATION "cube")
-    # Fix up the library references to be self-contained within the bundle.
-    install(CODE "
-        include(BundleUtilities)
-        fixup_bundle(\${CMAKE_INSTALL_PREFIX}/cube/vkcube.app \"\" \"\")
-        ")
+    install(
+        TARGETS vkcube
+        # Install runtime dependencies like the Vulkan::Loader so the app is self-contained
+        RUNTIME_DEPENDENCIES DESTINATION "cube/vkcube.app/Contents/Frameworks/"
+        BUNDLE DESTINATION "cube"
+    )
+    set_target_properties(vkcube PROPERTIES
+        INSTALL_RPATH @executable_path/../Frameworks
+    )
 else()
     install(TARGETS vkcube)
 endif()
@@ -300,14 +301,15 @@ target_include_directories(vkcubepp PRIVATE .)
 target_compile_definitions(vkcubepp PRIVATE VK_NO_PROTOTYPES)
 
 if(APPLE)
-    # Keep RPATH so fixup_bundle can use it to find libraries
-    set_target_properties(vkcubepp PROPERTIES INSTALL_RPATH_USE_LINK_PATH TRUE)
-    install(TARGETS vkcubepp BUNDLE DESTINATION "cube")
-    # Fix up the library references to be self-contained within the bundle.
-    install(CODE "
-        include(BundleUtilities)
-        fixup_bundle(\${CMAKE_INSTALL_PREFIX}/cube/vkcubepp.app \"\" \"\")
-        ")
+    install(
+        TARGETS vkcubepp
+        # Install runtime dependencies like the Vulkan::Loader so the app is self-contained
+        RUNTIME_DEPENDENCIES DESTINATION "cube/vkcubepp.app/Contents/Frameworks/"
+        BUNDLE DESTINATION "cube"
+    )
+    set_target_properties(vkcubepp PROPERTIES
+        INSTALL_RPATH @executable_path/../Frameworks
+    )
 else()
     install(TARGETS vkcubepp)
 endif()

